src/python_pachyderm/experimental/mixin/pfs.py

```diff
@@ -1254,6 +1258,85 @@ def path_exists(
 
         return True
 
+    def mount(self, mount_dir: str, repos: List[str] = []) -> None:
+        """Mounts Pachyderm repos locally.
+
+        Parameters
+        ----------
+        mount_dir : str
+            The directory to mount repos to. Make sure if this folder already
+            exists that it's empty (including hidden files).
+        repos : List[str], optional
+            The repos to mount. Each repo can only be mounted once, even if
+            multiple branches are passed. If empty, all repos are mounted.
+
+        Notes
+        -----
+        Mounting uses FUSE, which causes some known issues on macOS. For the
+        best experience, we recommend using mount on Linux. We do not support
+        mounting on macOS 1.11 and later.
+
+        Additionally, we recommend using mount in read-only access.
+
+        Examples
+        --------
+        >>> client.mount("dir_a", ["repo1", "repo2@staging"])
+        """
+        Path(mount_dir).mkdir(parents=True, exist_ok=True)
+
+        subprocess.run(
+            ["sudo", "pachctl", "unmount", mount_dir],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.STDOUT,
+        )
+
+        cmd = ["pachctl", "mount", mount_dir]
+        for r in repos:
+            cmd.append("-r")
+            cmd.append(r)
+
+        subprocess.Popen(cmd, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+        time.sleep(0.25)
+
+        # Check for successful mount
+        pach_repos = list(self.list_repo())
+        pach_repos = {repo_info.repo.name for repo_info in pach_repos}
+        repo_names = {r.split("@")[0] for r in repos}
```

Comment:
- I think it might make sense to do this check in a loop (in case pachctl mount is taking time to come up) and timeout/error only when N tries to read mounted data have failed
- Would it be possible to shorten this code by simplifying the success criteria to just look at empty/non-empty? Something like
```
for i in range(3):
  time.sleep(0.25)  # Give mount time to start
  try:
    next(os.walk(mount_dir))
    return  # Success
  except StopIteration:
    try:
      next(self.list_repo())
      return # Success
    except StopIteration
      continue
raise RuntimeError("pachctl mount failed to expose data after three read attempts (0.75s)")
self.unmount(repos...)
```

Reply:
Loop makes sense to me. Are you suggesting empty/non-empty because comparing repo names as I currently have it discovers the same conditions/outcomes?